Clean up site and remove outdated information

angelsl opened the pull request:
Closes #269. Closes #164 (the Code page is removed).

Things to do after this:

- Create a site/page for Hacker Tools
- Feature our publicity channels more prominently (#128, #59, etc)
- Slight reorganisation; currently some pages are only accessible via links in the middle of prose. Figure out somewhere to link this pages
  - Friday Hacks schedule (/fridayhacks/schedule)
  - Recordings
- Filter out Friday Hacks posts from the post listing in homepage (#54)

nushackers-org commented:
Deploy preview for *nushackers* ready!

Built with commit d15a65bbd7c5e3f1ff2109f58748350ed5ea0778

https://deploy-preview-271--nushackers.netlify.com

angelsl commented:
Summary of changes:

- Remove references to code.nushackers.org
- Remove separate Events page; the menu item now links to a section in About
- Feature Hacker Tools on the homepage (replacing code@NUS), although it currently does not link anywhere
- Change Discussion to Community, and update the links
- Update "What we do" in About, and the About Friday Hacks page
- Converted HTML to Markdown in the old ported single pages where possible
- Made the text sound a bit friendlier where possible
- Removed Title Case where it is not something copied from elsewhere
